c10/core/TensorImpl.cpp

```diff
@@ -184,6 +186,35 @@ TensorImpl::TensorImpl(
   // Caffe2 operators create Storages with default devices.
 }
 
+void TensorImpl::_change_backend_component_keys(c10::Device device) {
+  DispatchKey device_key =
+      computeDispatchKey(c10::nullopt, c10::nullopt, device);
```

Comment:
Feels like we should have a way to go directly from `Device` to `BackendComponent` - this feels a bit roundabout.